cmd/noms/splore/noms_splore.go

```diff
@@ -191,25 +191,47 @@ func getNodeChildren(v types.Value, parentPath string) (children []nodeChild) {
 			i := 0
 			v.IterAll(func(v types.Value) {
 				children[i] = nodeChild{
-					Value: info(v, atPath(i, "")),
+					Value: info(v, childPath("@at(%d)", i)),
 				}
 				i++
 			})
 		}
 	case types.Ref:
 		children = []nodeChild{{
-			Value: info(v, parentPath+"@target"),
+			Value: info(v, childPath("@target")),
 		}}
 	case types.Struct:
 		children = make([]nodeChild, v.Len())
 		i := 0
 		v.IterFields(func(name string, v types.Value) {
 			children[i] = nodeChild{
 				Label: name,
-				Value: info(v, fmt.Sprintf("%s.%s", parentPath, name)),
+				Value: info(v, childPath(".%s", name)),
 			}
 			i++
 		})
+	case *types.Type:
+		switch d := v.Desc.(type) {
+		case types.CompoundDesc:
+			children = make([]nodeChild, len(d.ElemTypes))
+			for i, t := range d.ElemTypes {
+				children[i] = nodeChild{
+					Value: info(t, childPath("[%d]", i)),
+				}
+			}
+		case types.StructDesc:
+			children = make([]nodeChild, d.Len())
+			i := 0
+			d.IterFields(func(name string, t *types.Type, optional bool) {
+				children[i] = nodeChild{
+					Label: name,
+					Value: info(t, childPath(".%s", name)),
+				}
+				i++
+			})
+		default:
+			children = []nodeChild{} // TODO: cycles?
```

Comment:
I would test this but the place I usually get cycles - from the photo data - is no longer compatible with this noms version. Will fix later.

Reply:
Cycles are represented as pointer cycles in memory so you would keep getting back to the same struct type over and over.

Reply:
It is easy enough to create a cyclic type programmatically of you want to test this.

Reply:
> you would keep getting back to the same struct 

Yep I see this with Commit. It's not a big deal. I think the graph library we use would get confused trying to draw a cyclic graph, so shrug

> want to test this 

Yeah, I'm just not sure what I want the behavior to be yet, I guess.